src/ophyd_async/epics/_backend/_aioca.py

```diff
@@ -35,7 +35,58 @@
     dbr.DBR_CHAR: "string",
     dbr.DBR_LONG: "integer",
     dbr.DBR_DOUBLE: "number",
+    dbr.DBR_ENUM: "string",
 }
+_num_dbr = {dbr.DBR_SHORT, dbr.DBR_CHAR, dbr.DBR_LONG, dbr.DBR_FLOAT, dbr.DBR_DOUBLE}
+_floating_dbr = {dbr.DBR_FLOAT, dbr.DBR_DOUBLE}
+
+
+def _data_key_from_augmented_value(source: str, value: AugmentedValue) -> DataKey:
+    """Use the return value of get with FORMAT_CTRL to construct a DataKey
+    describing the signal. See docstring of AugmentedValue for expected
+    value fields by DBR type.
+
+    Args:
+        value (AugmentedValue): Description of the the return type of a DB record
+
+    Returns:
+        DataKey: A rich DataKey describing the DB record
+    """
+
+    # Always present keys
+    if not value.ok:
+        raise Exception(f"{source} error: {value.errorcode}: {str(value)}")
+
+    scalar = value.element_count == 1
+
+    d = DataKey(
+        source=source,
+        dtype=dbr_to_dtype[value.datatype] if scalar else "array",
+        shape=[] if scalar else [len(value)],
+    )
+
+    if value.datatype in _num_dbr:
+        d.update(
+            {
+                "units": value.units,
+                "lower_alarm_limit": value.lower_alarm_limit,
+                "upper_alarm_limit": value.upper_alarm_limit,
+                "lower_ctrl_limit": value.lower_ctrl_limit,
+                "upper_ctrl_limit": value.upper_ctrl_limit,
+                "lower_disp_limit": value.lower_disp_limit,
+                "upper_disp_limit": value.upper_disp_limit,
+                "lower_warning_limit": value.lower_warning_limit,
+                "upper_warning_limit": value.upper_warning_limit,
+            }
+        )
+    if value.datatype in _floating_dbr:
+        d["precision"] = value.precision
+    if value.datatype is dbr.DBR_ENUM:
+        d["choices"] = value.enums
+    if value.datatype is dbr.DBR_STRING:
+        d["timestamp"] = value.timestamp
```

Comment:
Can we do `hasattr` here? There are some tricky upstream combinations of datatypes, but the value if present should always be put in the output